src/mon/OSDMonitor.cc

```diff
@@ -1663,6 +1665,12 @@ bool OSDMonitor::check_failure(utime_t now, int target_osd, failure_info_t& fi)
     for (map<int,failure_reporter_t>::iterator p = fi.reporters.begin();
 	 p != fi.reporters.end();
 	 ++p) {
+      //get subtree reporter
+      map<string,string> reporter_loc = osdmap.crush->get_full_location(p->first);
+      int osd_subtree_reporter  = osdmap.crush->get_item_id(reporter_loc[reporter_subtree_level]);
+      assert(osd_subtree_reporter != 0);
+      subtree_reporters.insert(osd_subtree_reporter);
```

Comment:
Ah, yeah.. fair enough.  But we still need to be more tolerant of the osd_subtree_reporter result.  (For example, if the subtree_level is osd then a 0 is a valid return value.)